### PR TITLE
Phase 3 Task 14 cycle 1: MulticolLayouter Hello World

### PR DESCRIPTION
**Pull request by raroche**
## Summary

Task 13 (multi-page row splitting + `<thead>`/`<tfoot>` repetition + 2 hardening passes) closed out the table layout pipeline. Task 14 opens **CSS Multi-column Layout L1** ([spec](https://www.w3.org/TR/css-multicol-1/)). Cycle 1 ships **Hello World multi-column layout**: a block container with `column-count: N` splits its in-flow content equally across N parallel columns, each acting as a sub-fragmentainer.

Continues from PR #55 merged at `d1157b3`.

## New types

- **`MulticolLayouter`** (`src/NetPdf.Layout/Layouters/MulticolLayouter.cs`, 489 LOC) — `internal sealed class : ILayouter, IDisposable`. Constructor mirrors `BlockLayouter`/`TableLayouter`. Validates `rootBox.Style.ReadColumnCount() >= 2` or throws.
- **`ColumnFragmentSink`** (internal nested decorator) — translates each emitted fragment's `InlineOffset` by `c * (columnInlineSize + columnGap)`.
- **`MulticolContinuation`** (reserved in `LayoutContinuation.cs` for sub-cycle 2 multi-page multicol).

## Algorithm

1. Read `column-count: N` (clamp `[1, 1000]`) + `column-gap: <length>` (cycle 1 hard-codes 16px; `normal`-keyword cascade resolution is sub-cycle 2).
2. Per-column inline-size = `(containerInlineSize - (N-1) * columnGap) / N`.
3. For each column `c` in `[0, N)`:
   - Construct sub-`FragmentainerContext` (per-column block-size = container content-block-size).
   - Construct nested `BlockLayouter` with `rootBox = multicol container` + a `ColumnFragmentSink` wrapping the outer sink with the column's `inlineOffsetTranslation`.
   - Capture `PageComplete(BlockContinuation)` from previous column → pass as `incomingContinuation` to next column.
4. Outer multicol box: emit ONE `BoxFragment` at parent's allocated geometry; column content lands INSIDE.
5. **Overflow simplification**: content overflowing N columns emits new `LAYOUT-MULTICOL-FORCED-OVERFLOW-001` Warning + truncates remaining content. **Multi-page multicol** (outer fragmented across pages) is **sub-cycle 2** work via the reserved `MulticolContinuation`.

## BlockLayouter integration

- Dispatch in **BOTH** the main `AttemptLayout` child-emit loop AND `EmitBlockSubtreeRecursive` (recursion needed because real HTML `<body><div class="multicol">...</div></body>` reaches multicol via the body-wrap recursion — verified by the production test, which would emit full-width content otherwise).
- `column-count: 1` bypasses (predicate gates on `>= 2`); content lays out as normal block flow.

## CSS property pipeline

- `column-count` / `column-width` / `column-gap` / `column-fill` / `column-rule-{width,style,color}` added to `src/NetPdf.Css/properties.json`.
- New per-property keyword paths in `KeywordResolver` + `LengthResolver` + `NumberResolver` for `auto` / `normal` defaults (previously hit `CSS-PROPERTY-VALUE-INVALID-001`).
- New `ReadColumnCount(this ComputedStyle) → int?` + `ReadColumnGap(this ComputedStyle) → double` extensions.
- `column-gap` removed from `RealDocuments` `UnsupportedAllowlist`.

## New diagnostic

`LAYOUT-MULTICOL-FORCED-OVERFLOW-001` (Warning) added to `PaginateDiagnosticCodes` + facade + `docs/diagnostics-codes.md` + 3 parity tests.

## Files changed

3 new files + 14 modified + 1 docs:

- **NEW** `src/NetPdf.Layout/Layouters/MulticolLayouter.cs` (489)
- **NEW** `tests/NetPdf.UnitTests/Phase3/MulticolLayouterTests.cs` (590)
- **NEW** `tests/NetPdf.UnitTests/Phase3/MulticolLayouterProductionTests.cs` (233)
- `src/NetPdf.Layout/Layouters/BlockLayouter.cs` (+201)
- `src/NetPdf.Layout/Layouters/ComputedStyleLayoutExtensions.cs` (+44)
- `src/NetPdf.Paginate/LayoutContinuation.cs` (+32)
- `src/NetPdf.Paginate/Diagnostics/PaginateDiagnosticCodes.cs` (+21)
- `src/NetPdf/Diagnostics/DiagnosticCodes.cs` (+19)
- `src/NetPdf.Css/properties.json` (+7)
- `src/NetPdf.Css/ComputedValues/PropertyResolvers/{Length,Number,Keyword}Resolver.cs` (+72 across all three)
- `tests/NetPdf.UnitTests/Diagnostics/DiagnosticCodesTests.cs` (+31, 3 parity tests)
- `tests/NetPdf.UnitTests/Docs/DeferralsParityTests.cs` (+1, new deferral ID)
- `tests/NetPdf.RealDocuments/Css/PropertyCoverageCorpusTests.cs` (`column-gap` removed from allowlist)
- `docs/deferrals.md` (+91, new `multicol-balancing-pagination` entry)
- `docs/diagnostics-codes.md` (+3, new code row)
- `PROGRESS.md` — Task 14 cycle 1 entry.

Net diff: **+1843 / −4** across 18 files.

## Tests

**4323 unit / 4454 solution-wide passing** (+15 unit / +15 solution-wide over PR #55 baseline at 4308 / 4439 — 11 direct tests + 1 production test + 3 diagnostic-codes parity; 4 skipped). AOT/JIT byte-parity gate stays green; `git diff --check` clean.

New direct tests in `MulticolLayouterTests.cs`:

1. 2-column emit + outer wrapper present
2. Equal-split minus gaps geometry
3. Content distribution across columns
4. Overflow → forced-overflow diagnostic
5. `column-count: 1` bypasses multicol
6. Rejects invalid column counts (zero/negative)
7. Default gap = 16px
8. Empty container emits outer only
9. Cancellation observed
10. + 2 more pinning fragment count + ordering

New production test in `MulticolLayouterProductionTests.cs`: `<div style="column-count: 2; width: 200px;">` lays out content into 2 columns end-to-end through the full HTML → BoxBuilder → BlockLayouter pipeline.

## Sub-cycle 2+ scope

Per the new `multicol-balancing-pagination` deferral entry:

- **Multi-page multicol** (outer fragmented across pages via the reserved `MulticolContinuation`)
- Column balancing (`column-fill: balance`)
- `column-width`-derived automatic column count
- `column-span: all`
- Column rules paint (cycle 1 parses the cascade but doesn't paint)
- `column-gap: normal` resolving to 1em via cascaded font-size (cycle 1 hard-codes 16 px)
- Fragmentation interaction with `avoid-break`
- Per-column resolver isolation for `break-inside: avoid-column`

## Test plan

- [x] `dotnet build NetPdf.slnx -c Release` — 0 warnings, 0 errors
- [x] `dotnet test NetPdf.slnx -c Release` — 4323 unit / 4454 solution-wide passing
- [x] `./scripts/aot-parity.sh` — AOT/JIT byte-parity verified
- [x] `git diff --check` — clean (no trailing whitespace)
- [ ] Roland review cycle on this PR
- [ ] Address Copilot findings if any

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/NetPdf.Css/ComputedValues/PropertyResolvers/LengthResolver.cs [hunk 1]
```diff
@@ -70,6 +70,19 @@ public static ResolverResult Resolve(
...
+        // Per Phase 3 Task 14 cycle 1 — per-property keyword admittance
+        // for the multicol family. CSS Multi-column L1 admits:
+        //   - column-width: <length> | auto
+        //   - column-gap:   <length> | normal
+        // Both are Length-typed (not LengthPercentageAuto / TextSpacing)
+        // because they don't admit percentages OR the broader keyword
+        // sets. The cycle-1 layouter reads these via
+        // <c>ComputedStyleLayoutExtensions.ReadColumnGap</c> /
+        // <c>ReadColumnWidth</c> which treat a non-LengthPx slot as the
+        // initial keyword.
```
[the remark this answers is not in the document]
Comment (raroche) [reply src/NetPdf.Css/ComputedValues/PropertyResolvers/LengthResolver.cs]: Resolved in (this hardening commit) as part of Roland's Copilot stale-docs cleanup. The XML doc at MulticolLayouter.cs:82 incorrectly referenced `ReadColumnWidth` (which doesn't exist as a layout-side extension); corrected to `ReadColumnGap` / `ReadColumnCount` (the two extensions cycle 1 actually provides). `column-width` reading lands when sub-cycle 2 implements width-derived auto column count.

File: src/NetPdf.Paginate/Diagnostics/PaginateDiagnosticCodes.cs [hunk 2]
```diff
@@ -179,4 +179,25 @@ internal static class PaginateDiagnosticCodes
...
+    /// available block-size. Cycle 1 ships Hello World multi-column
+    /// layout: a block container with <c>column-count: N</c> splits
+    /// its content equally across N parallel columns (sub-
+    /// fragmentainers), each sized at
+    /// <c>(containerContentInlineSize - (N-1)*columnGap) / N</c>
+    /// inline-axis + <c>containerContentBlockSize</c> block-axis.
+    /// When content overflows the FIRST column the layouter advances
+    /// to the next; when content overflows the LAST column the
+    /// remaining content is truncated + this diagnostic fires. Per
+    /// CSS Multi-column L1 §3.5 the spec-strict behavior is to
```
[the remark this answers is not in the document]
Comment (raroche) [reply src/NetPdf.Paginate/Diagnostics/PaginateDiagnosticCodes.cs:195]: Resolved in (this hardening commit). The XML doc at MulticolLayouter.cs:195 had inaccurate wording — said "When content overflows the FIRST column the layouter advances to the next" but the implementation advances when ANY non-last column overflows (= when column `c < N-1` returns PageComplete, the next column's nested BlockLayouter is constructed with that continuation as `incomingContinuation`). Doc rewritten to describe the actual non-last-column overflow semantics.

File: docs/diagnostics-codes.md [hunk 3]
```diff
@@ -118,6 +118,7 @@ Severity levels:
...
+| `LAYOUT-MULTICOL-FORCED-OVERFLOW-001` | Warning | The in-flow content of a multicol container (block container with `column-count: N`) does NOT fit within the N columns' available block-size. Cycle 1 ships Hello World multi-column layout: a block container with `column-count: N` splits its content equally across N parallel columns, each acting as a sub-fragmentainer of the same block-axis extent. When content overflows the LAST column the remaining content is truncated + this diagnostic fires. Per CSS Multi-column L1 §3.5 the spec-strict behavior is to fragment the multicol container itself across pages so the overflowing content continues; sub-cycle 2 will ship that multi-page multicol via `MulticolContinuation`. The diagnostic message names the column count + the per-column block-size + the leftover content extent. See `docs/deferrals.md#multicol-balancing-pagination`. Per Phase 3 Task 14 cycle 1. |
```
[the remark this answers is not in the document]
Comment (raroche) [reply docs/diagnostics-codes.md]: Resolved in (this hardening commit). The `docs/diagnostics-codes.md:121` row for `LAYOUT-MULTICOL-FORCED-OVERFLOW-001` claimed the emitted message names "the leftover content extent" but the actual message in `MulticolLayouter` didn't include that. Doc aligned with the actual emitted message — describes what the diagnostic actually says (overflow column count + container's available extent).

File: PROGRESS.md [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
... bypasses multicol; rejects invalid column counts (zero/negative); default gap = 16px; empty container emits outer only; cancellation observed; + 2 more pinning fragment count + ordering. New production test in [`MulticolLayouterProductionTests`](tests/NetPdf.UnitTests/Phase3/MulticolLayouterProductionTests.cs): `<div style="column-count: 2; width: 200px;">` lays out content into 2 columns end-to-end through the full HTML → BoxBuilder → BlockLayouter pipeline. **Sub-cycle 2+ scope:** multi-page multicol (outer fragmented across pages via the reserved `MulticolContinuation`); column balancing (`column-fill: balance`); `column-width`-derived automatic column count; `column-span: all`; column rules paint (cycle 1 parses the cascade but doesn't paint); `column-gap: normal` resolving to 1em via cascaded font-size (cycle 1 hard-codes 16 px); fragmentation interaction with `avoid-break`; per-column resolver isolation for `break-inside: avoid-column`. Active-task pointer advances to **Task 14 cycle 2** — column balancing OR multi-page multicol, depending on review priority. Plus all remaining Task 12 sub-cycle 6+ items + Task 13 deeper-nesting recursion-continuation propagation deferrals.)
```
[the remark this answers is not in the document]
Comment (raroche) [reply PROGRESS.md]: Resolved in (this hardening commit) as Copilot #4. The PROGRESS.md entry now accurately describes the contract: the **constructor's** predicate is `>= 1` (so direct-construction tests of the `column-count: 1` edge case work + future sub-cycles with width-derived counts that round to 1 can construct a layouter); the **BlockLayouter dispatch gate** that decides whether to invoke MulticolLayouter at all is the stricter `>= 2` — `column-count: 1` lays out as a normal block. Two distinct contracts, now spelled out explicitly.
